src/matcha_ml/core/core.py

```diff
@@ -90,3 +97,80 @@ def remove_state_lock() -> None:
     """Unlock remote state."""
     remote_state = RemoteStateManager()
     remote_state.unlock()
+
+
+@track(event_name=AnalyticsEvent.PROVISION)
+def provision(
+    location: str,
+    prefix: str,
+    password: str,
+    verbose: Optional[bool] = False,
+) -> MatchaState:
+    """Provision cloud resources using templates.
+
+    Args:
+        location (str): Azure location in which all resources will be provisioned.
+        prefix (str): Prefix used for all resources.
+        password (str): Password for ZenServer.
+        verbose (bool optional): additional output is show when True. Defaults to False.
+
+    Returns:
+        MatchaState: the information of the provisioned resources.
+
+    Raises:
+        MatchaError: If resources are already provisioned.
+        MatchaError: If prefix is not valid.
+        MatchaError: If region is not valid.
+    """
+    remote_state_manager = RemoteStateManager()
+
+    if remote_state_manager.is_state_stale():
+        print_status(
+            build_warning_status(
+                "Matcha has detected a stale state file. This means that your local configuration is out of sync with the remote state, the resource group may have been removed. Deleting existing state config."
+            )
+        )
+        remote_state_manager.remove_matcha_config()
+
+    if remote_state_manager.is_state_provisioned():
+        raise MatchaError(
+            "Error - Matcha has detected that there are resources already provisioned. Use 'matcha destroy' to remove the existing resources before trying to provision again."
+        )
+
+    # Input checks
+    try:
+        prefix = prefix.lower()
+        _ = is_valid_prefix(prefix)
+        _ = is_valid_region(location)
+    except MatchaInputError as e:
+        raise e
+
+    # Provision resource group and remote state storage
+    remote_state_manager.provision_remote_state(location, prefix)
+
+    with remote_state_manager.use_lock(), remote_state_manager.use_remote_state():
+        # create a runner for provisioning resource with Terraform service.
+        template_runner = AzureRunner()
+
+        project_directory = os.getcwd()
+        destination = os.path.join(
+            project_directory, ".matcha", "infrastructure", "resources"
+        )
+        template = os.path.join(
+            os.path.dirname(__file__), os.pardir, "infrastructure", "resources"
+        )
+
+        # Create a azure template object
```

Comment:
Nit: these types of comments don't add anything, the code is descriptive enough (I appreciate that it was probably brought over from existing code)